temple/utils.py

```diff
@@ -126,3 +126,30 @@ def _call_api(self, verb, url, **request_kwargs):
     def get(self, url, **request_kwargs):
         """Github API get"""
         return self._call_api('get', url, **request_kwargs)
+
+class GitLabClient():
+    def __init__(self, template):
+        api_token = os.environ.get('GITLAB_API_TOKEN')
+        gitlab_api = 'https://gitlab.com/api/v4'
+        self.headers = { 'Authorization': f"Bearer {api_token}" }
+        query = template.split('/')[-1:]
+        url = f"{gitlab_api}/projects?search={query}"
+        search_response = requests.get(url, headers=self.headers)
```

Comment:
Doing a get request in the init method of the class is unusual. Also why do we need to search for the project? We have the template URL right here.